runtime/src/main/java/io/quarkiverse/freemarker/runtime/FreemarkerBuildConfig.java

```diff
@@ -15,16 +19,120 @@
      * Comma-separated list of absolute resource paths to scan recursively for templates.
      * All tree folder from 'resource-paths' will be added as a resource.
      * Unprefixed locations or locations starting with classpath will be processed in the same way.
+     *
+     * @deprecated Use template set group of options instead ({@code quarkus.freemarker.<template-set-name>.base-path}
+     *             {@code quarkus.freemarker.<template-set-name>.includes} and
+     *             {@code quarkus.freemarker.<template-set-name>.excludes})
      */
-    @ConfigItem(defaultValue = "freemarker/templates")
-    public List<String> resourcePaths;
+    @ConfigItem
+    public Optional<List<String>> resourcePaths;
 
     /**
      * List of directives to register with format name=classname
-     * 
+     *
      * @see freemarker.template.Configuration#setSharedVariable(String, TemplateModel)
      */
     @ConfigItem
     public Map<String, String> directive;
 
+    /**
+     * The default template set
+     */
+    @ConfigItem(name = ConfigItem.PARENT)
+    public TemplateSet defaultTemplateSet;
+
+    /**
+     * Additional named template sets.
+     */
+    @ConfigDocSection
+    @ConfigDocMapKey("template-set-name")
+    @ConfigItem(name = ConfigItem.PARENT)
+    public Map<String, TemplateSet> namedTemplateSets;
+
+    @ConfigGroup
+    public static class TemplateSet {
+        /**
+         * The base path of this template set.
+         * <p>
+         * Use slash ({@code /}) as a path separator on all platforms. The value must not start with a slash.
+         * <p>
+         * {@link #includes} and {@link #excludes} are relative to {@link #basePath}.
+         * <p>
+         * Defaults:
+         * <table>
+         * <tr>
+         * <th>Option</th>
+         * <th>Default value</th>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.freemarker.base-path}</td>
+         * <td>{@code freemarker/templates}</td>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.freemarker.includes}</td>
+         * <td>{@code **}</td>
```

Comment:
Yes, my intention was to avoid an unwanted inclusion of all resources when the user sets a basePath but omits includes (which may substantially increase the size of the native image). Maybe that case should result in an error. WDYT?

Reply:
In 24db477, I made that case a build time error.

Reply:
I am actually fine either way. we just need the javadoc to be consistent with the behavior. what do you mean by `**`?  I assumed _everything_. and what I am understanding from the `toBuildItem` impl is that if we omit the `includes`, we won't be including anything, which is not what the comment is saying.

so if the javadoc says `**`, I would expect:
```
        @ConfigItem(defaultValue = "**")
        public List<String> includes;
```

but if we want to require this config property, then we should not make it optional, and update the javadoc accordingly.
I am actually fine with `includes` being by default `**` or `**.ftl`, as your documenting later. this cover the case where you want everything with some exclusions.
BTW, where did you make it a build time error? 
am I missing something?

Reply:
Thanks for the feedback. Yes, the JavaDoc and code are a bit confusing.

The defaults for `basePath` and `includes` are set dynamically (rather than via `@ConfigItem(defaultValue = ...)`) because my intention was to have them only to achieve the backwards compatibility with the original `resourcePaths`. 

The current implementation applies the default to `basePath` and `includes` only if none of `quarkus.freemarker.base-path`, `quarkus.freemarker.includes` and `quarkus.freemarker.excludes` is set. It happens around [here](https://github.com/quarkiverse/quarkus-freemarker/pull/45/files#diff-05faee15c7e8258176aa2332fc13674b4975c3fa539f08438804a2d69118fecfR66-R69).

I'd prefer making the JavaDoc consistent with the code rather than vice versa. I should also explain the motivation in the JavaDoc.